src/main/java/graphql/annotations/DefaultTypeFunction.java

```diff
@@ -59,6 +55,25 @@ void setAnnotationsProcessor(GraphQLAnnotationsProcessor annotationsProcessor) {
         this.annotationsProcessor = annotationsProcessor;
     }
 
+    private class IDFunction implements TypeFunction {
+        @Override
+        public String getTypeName(Class<?> aClass, AnnotatedType annotatedType) {
+            return Scalars.GraphQLID.getName();
+        }
+
+        @Override
+        public boolean canBuildType(Class<?> aClass, AnnotatedType annotatedType) {
+            return annotatedType != null
+                    && (aClass == Integer.class || aClass == int.class || aClass == String.class)
```

Comment:
Why do you restrict the ID to be a String or an int? What If I want my ID to be a costum type? (UUID for example)

Reply:
Because the spec defines GraphQLID to be integer/string, it can only accept those 2 types:

"When expected as an input type, any string (such as "4") or integer (such as 4) input value should be coerced to ID as appropriate for the ID formats a given GraphQL server expects. Any other input value, including float input values (such as 4.0), must raise a query error indicating an incorrect type."